packages/p2p/src/components/modal-manager/modal-manager-context-provider.jsx

```diff
@@ -45,24 +45,45 @@ const ModalManagerContextProvider = props => {
         }
         setIsModalOpen(true);
     };
-    const hideModal = (should_save_form_history = false) => {
+
+    /**
+     * Hides the current shown modal.
+     * If a previous modal was present, by default the previous modal will be shown in-place of the current closed modal.
+     * This option can be overriden by setting `hide_all_modals` to `true` in the `options` argument to close all modals instead.
+     *
+     * @param {Object} options - list of supported settings to tweak how modals should be hidden:
+     * - **hide_all_modals**: `false` by default. If set to `true`, previous modal will not be shown and all modals are hidden.
+     * - **save_form_history**: `false` by default. If set to `true`, form values in modals that has a form with `ModalForm` component
+     * will be saved when the modal is hidden and restored when modal is shown again.
+     */
+    const hideModal = (options = {}) => {
+        const { save_form_history = false, hide_all_modals = false } = options;
+      
         modal_props.delete(active_modal.key);
         if (isDesktop()) {
-            if (should_save_form_history) {
+            if (save_form_history) {
                 general_store.saveFormState();
             } else {
                 general_store.setSavedFormState(null);
                 general_store.setFormikRef(null);
             }
 
-            if (previous_modal) {
+            if (hide_all_modals) {
+                setPreviousModal({});
+                setActiveModal({});
+                setIsModalOpen(false);
+            } else if (previous_modal) {
                 setActiveModal(previous_modal);
-                setPreviousModal(null);
+                setPreviousModal({});
             } else {
                 setActiveModal({});
                 setIsModalOpen(false);
             }
```

Comment:
is it possible to just provide the key to hide instead of having these conditions?

Reply:
These conditions are needed because we need to store and check if the previous modal object exists when another modal needs to be shown, because that previous modal that was stored could be shown again once the current modal is hidden